hermes/src/store.rs

```diff
@@ -127,7 +127,11 @@ impl Store {
                     }
                 };
 
-                self.observed_vaa_seqs.insert(vaa.sequence, true).await;
+                let mut observed_vaa_seqs = self.observed_vaa_seqs.write().await;
+                observed_vaa_seqs.insert(vaa.sequence);
+                if observed_vaa_seqs.len() > OBSERVED_CACHE_SIZE {
+                    observed_vaa_seqs.pop_first();
+                }
```

Comment:
i presume that we drop the lock as soon as the mutable ref goes out of scope? if so, should this bit go in a { } block so that we drop the lock as soon as we can?

Reply:
nice catch!